File: openadmin/spec/area_chart.py
# SPDX-FileCopyrightText: 2026 OpenAdmin
#
# SPDX-License-Identifier: AGPL-3.0-or-later

from typing import Literal, TypedDict

from .http_methods import HttpMethod
from .json_schema import JsonSchema


class AreaChart(TypedDict):
    type: Literal["area-chart"]
    id: str
    name: str
    description: str | None
    method: HttpMethod
    form: JsonSchema | None
    body: JsonSchema | None
    query: JsonSchema | None

File: client/src/schemas/area-chart.ts
// SPDX-FileCopyrightText: 2026 OpenAdmin
//
// SPDX-License-Identifier: AGPL-3.0-or-later

import { z } from "zod"
import { httpMethodSchema } from "./http-method"
import { jsonSchemaSchema } from "./json-schema"

export const areaChartComponentSchema = z.object({
	type: z.literal("area-chart"),
	id: z.string(),
	name: z.string(),
	description: z.string().nullable(),
	refresh: z.number().nullable(),
	method: httpMethodSchema,
	form: jsonSchemaSchema.nullable(),
	body: jsonSchemaSchema.nullable(),
	query: jsonSchemaSchema.nullable(),
})

export type AreaChartComponent = z.infer<typeof areaChartComponentSchema>

export const areaChartSchema = areaChartComponentSchema

export type AreaChart = AreaChartComponent
